x/monitoringc/keeper/monitoring_packet_test.go

```diff
@@ -0,0 +1,139 @@
+		coins          = sample.Coins(r)
+	)
+
+	tk.MonitoringConsumerKeeper.SetLaunchIDFromChannelID(ctx, types.LaunchIDFromChannelID{
+		ChannelID: invalidChannel,
+		LaunchID:  10000,
+	})
+	chain.LaunchID = tk.LaunchKeeper.AppendChain(ctx, chain)
+	tk.MonitoringConsumerKeeper.SetLaunchIDFromChannelID(ctx, types.LaunchIDFromChannelID{
+		ChannelID: validChannel,
+		LaunchID:  chain.LaunchID,
+	})
+
+	tk.RewardKeeper.SetRewardPool(ctx, rewardtypes.RewardPool{
+		LaunchID:         chain.LaunchID,
+		Provider:         sample.Address(r),
+		InitialCoins:     coins,
+		RemainingCoins:   coins,
+		LastRewardHeight: 1,
+		Closed:           false,
+	})
+	err := tk.BankKeeper.MintCoins(ctx, rewardtypes.ModuleName, coins)
+	require.NoError(t, err)
+
+	// set validator profiles
+	tk.ProfileKeeper.SetValidator(ctx, profiletypes.Validator{
+		Address:           valFoo,
+		OperatorAddresses: []string{valOpAddrFoo},
+	})
+	tk.ProfileKeeper.SetValidatorByOperatorAddress(ctx, profiletypes.ValidatorByOperatorAddress{
+
+		ValidatorAddress: valFoo,
+		OperatorAddress:  valOpAddrFoo,
+	})
+	tk.ProfileKeeper.SetValidator(ctx, profiletypes.Validator{
+		Address:           valBar,
+		OperatorAddresses: []string{valOpAddrBar},
+	})
+	tk.ProfileKeeper.SetValidatorByOperatorAddress(ctx, profiletypes.ValidatorByOperatorAddress{
+		ValidatorAddress: valBar,
+		OperatorAddress:  valOpAddrBar,
+	})
+
+	tests := []struct {
+		name   string
+		packet channeltypes.Packet
+		data   spntypes.MonitoringPacket
+		valid  bool
+	}{
+		{
+			name:   "invalid data",
+			packet: channeltypes.Packet{},
+			data: spntypes.MonitoringPacket{
+				BlockHeight: 0,
+				SignatureCounts: spntypes.SignatureCounts{
+					BlockCount: 1,
+				},
+			},
+			valid: false,
+		},
+		{
+			name: "no launch ID associated to channel ID",
+			packet: channeltypes.Packet{
+				DestinationChannel: "invalid",
+			},
+			data: spntypes.MonitoringPacket{
+				BlockHeight: 1,
+				SignatureCounts: spntypes.SignatureCounts{
+					BlockCount: 1,
+				},
+			},
+			valid: false,
+		},
+		{
+			name: "should fail distribute rewards",
+			packet: channeltypes.Packet{
+				DestinationChannel: invalidChannel,
+			},
+			data: spntypes.MonitoringPacket{
+				BlockHeight: 1,
+				SignatureCounts: spntypes.SignatureCounts{
+					BlockCount: 1,
+				},
+			},
+			valid: false,
+		},
+		{
+			name: "should successfully distribute rewards",
+			packet: channeltypes.Packet{
+				DestinationChannel: validChannel,
+			},
+			data: spntypes.MonitoringPacket{
+				BlockHeight: 10,
+				SignatureCounts: tc.SignatureCounts(10,
+					tc.SignatureCount(t, valOpAddrFoo, "0.5"),
+					tc.SignatureCount(t, valOpAddrBar, "0.5"),
+				),
+			},
+			valid: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := tk.MonitoringConsumerKeeper.OnRecvMonitoringPacket(ctx, tt.packet, tt.data)
+			if !tt.valid {
+				require.Error(t, err)
+				return
+			}
+			require.NoError(t, err)
+		})
+	}
+}
```

Comment:
We can also add test cases to ensure `OnAcknowledgementMonitoringPacket` and `OnTimeoutMonitoringPacket` return errors